src/widgets/osd.cpp

```diff
@@ -291,57 +271,74 @@ void OSD::RepeatModeChanged(PlaylistSequence::RepeatMode mode) {
   }
 }
 
-QString OSD::ReplaceVariable(const QString &variable, const Song &song) {
+QString OSD::ReplaceMessage(const QString &message, const Song &song, OSD::Behaviour behaviour) {
+
+  QRegExp variable_replacer("[%][a-z]+[%]");
+  QString copy(message);
+
+  // Replace the first line
+  int pos = 0;
+  variable_replacer.indexIn(message);
+  while ((pos = variable_replacer.indexIn(message, pos)) != -1) {
+    QStringList captured = variable_replacer.capturedTexts();
+    copy.replace(captured[0], ReplaceVariable(captured[0], song, behaviour));
+    pos += variable_replacer.matchedLength();
+  }
+
+  return copy;
+}
+
+QString OSD::ReplaceVariable(const QString &variable, const Song &song, OSD::Behaviour behaviour) {
```

Comment:
Maybe we could move ReplaceVariable to utilities, and keep the OSD::Behaviour stuff in it's own function in  the osd.